python/lsst/verify/ap/dataset.py

```diff
@@ -0,0 +1,218 @@
+
+        return Dataset._dataset_config
+
+    @property
+    def dataset_root(self):
+        """The parent directory containing everything related to the dataset.
+
+        Returns
+        -------
+        a string giving the location of the base directory
+        """
+        return self._data_root_dir
+
+    @property
+    def data_location(self):
+        """The directory containing the "raw" input data.
+
+        Returns
+        -------
+        a string giving the location of the top-level directory for telescope output files
+        """
+        return os.path.join(self.dataset_root, 'raw')
+
+    @property
+    def calib_location(self):
+        """The directory containing the calibration data.
+
+        Returns
+        -------
+        a string giving the location of the top-level directory for master calibration files
+        """
+        return os.path.join(self.dataset_root, 'calib')
+
+    @property
+    def template_location(self):
+        """The directory containing the image subtraction templates.
+
+        Returns
+        -------
+        a string giving the location of the top-level directory for precomputed templates
+        """
+        return os.path.join(self.dataset_root, 'templates')
+
+    @property
+    def _stub_input_repo(self):
+        """The directory containing the data set's input stub.
+
+        Returns
+        -------
+        a string giving the location of the stub input repo
+        """
+        return os.path.join(self.dataset_root, 'data')
+
+    def _validate_package(self):
+        """Confirm that the dataset directory satisfies all assumptions.
+
+        Requires that self._data_root_dir has been initialized.
+
+        Raises
+        ------
+        `RuntimeError`:
+            if any problems are found with the package
+        """
+        if not os.path.exists(self.dataset_root):
+            raise RuntimeError('Could not find dataset at ' + self.dataset_root)
+        if not os.path.exists(self.data_location):
+            raise RuntimeError('Dataset at ' + self.dataset_root + 'is missing data directory')
+        if not os.path.exists(self.calib_location):
+            raise RuntimeError('Dataset at ' + self.dataset_root + 'is missing calibration directory')
+        # Template directory might not be subdirectory of self.dataset_root
+        if not os.path.exists(self.template_location):
+            raise RuntimeError('Dataset is missing template directory at ' + self.template_location)
+        if not os.path.exists(self._stub_input_repo):
+            raise RuntimeError('Dataset at ' + self.dataset_root + 'is missing stub repo')
+        if not os.path.exists(os.path.join(self._stub_input_repo, '_mapper')):
+            raise RuntimeError('Stub repo at ' + self._stub_input_repo + 'is missing mapper file')
+
+    def make_output_repo(self, output_dir):
+        """Set up a directory as an output repository compatible with this dataset.
+
```

Comment:
Reruns are translated to directories in the `verify_ap` module, so this class doesn't need to care.